_includes/manuals/1.19/4.2.5_param_classes.md

```diff
@@ -0,0 +1,90 @@
+
+Parameterized class support permits detecting, importing, and supplying parameters direct to classes which support it, via the ENC.  This requires Puppet 2.6.5 or higher.
+
+#### Setup
+
+By default, parameterized class support is enabled in Foreman.  This can be checked from *Administer > Settings > Puppet* and ensure *Parametrized_Classes_in_ENC* is set to *true*.
+
+![Settings](/static/images/screenshots/param_classes/settings.png)
+
+Now you'll need to import some parameterized classes from your Puppet master.
+If you don't have any parameterized classes in your modules dir, the
+foreman-installer has several, you can download a few modules from the Puppet
+Forge. Once you have some parameterized modules, import your classes (see
+[**4.2.2 Classes**](manuals/{{page.version}}/index.html#4.2.2Classes))
+
+![Import](/static/images/screenshots/param_classes/1.17/import.png)
+
+#### Configure a class
+
+This example will work with the *foreman* class from the installer. Click on the class, and you should get a page with 3 tabs, like so:
+
+![3 Tabs](/static/images/screenshots/param_classes/3tabs.png)
+
+The middle tab, "Smart Class Parameter", is the important one. Click onto that, and you should see something like this:
+
+![Edit](/static/images/screenshots/param_classes/1.14/edit.png)
+
+On the left, we have a list of possible parameters that the class supports. On the right, we have the configuration options for the parameter selected.
+
+Lets configure the *foreman* class to change the user the foreman processes run as. Select the *user* parameter, at the end of the list. Now lets go through the options:
```

Comment:
This list isn't entirely correct anymore. The order has changed:
* Key (added)
* Description
* Puppet Environments
* Override
* Key type
* Default value
* Omit / Hidden value